src/components/PDFView/PDFPasswordForm.js

```diff
@@ -45,133 +41,121 @@ const defaultProps = {
     onPasswordFieldFocused: () => {},
 };
 
-class PDFPasswordForm extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            password: '',
-            validationErrorText: '',
-            shouldShowForm: false,
-        };
-        this.submitPassword = this.submitPassword.bind(this);
-        this.updatePassword = this.updatePassword.bind(this);
-        this.showForm = this.showForm.bind(this);
-        this.validateAndNotifyPasswordBlur = this.validateAndNotifyPasswordBlur.bind(this);
-        this.getErrorText = this.getErrorText.bind(this);
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.isFocused || !this.props.isFocused || !this.textInputRef) {
-            return;
-        }
-        this.textInputRef.focus();
-    }
+function PDFPasswordForm(props) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {translate} = useLocalize();
 
-    getErrorText() {
-        if (this.props.isPasswordInvalid) {
-            return this.props.translate('attachmentView.passwordIncorrect');
+    const [password, setPassword] = useState('');
+    const [validationErrorText, setValidationErrorText] = useState('');
+    const [shouldShowForm, setShouldShowForm] = useState(false);
+    const textInputRef = useRef(null);
+
+    useEffect(() => {
+        if (!props.isFocused || !textInputRef.current) return;
+        textInputRef.current.focus();
+    }, [props.isFocused]);
+
+    const getErrorText = () => {
+        if (props.isPasswordInvalid) {
+            return translate('attachmentView.passwordIncorrect');
         }
-        if (!_.isEmpty(this.state.validationErrorText)) {
-            return this.props.translate(this.state.validationErrorText);
+        if (!_.isEmpty(validationErrorText)) {
+            return translate(validationErrorText);
         }
 
         return '';
-    }
+    };
 
-    submitPassword() {
-        if (!this.validate()) {
-            return;
+    const updatePassword = (passwrd) => {
```

Comment:
```suggestion
    const updatePassword = (newPassword) => {
```
NAB `passwrd` looks like a typo, it may be confusing with `password`. `newPassword` looks more as an intended change.